src/main/java/frc/robot/vision/VisionCommManager.java

```diff
@@ -0,0 +1,36 @@
+package frc.robot.vision;
+
+import frc.robot.utils.JStruct;
+
+public class VisionCommManager{
+    private SerialProvider connection;
+
+    private DataFrame latestTargetData;
+    private DataFrame latestImage;
+    private JStruct struct = new JStruct();
+
+    class DataFrame{
+        public char type;
```

Comment:
not a fan of having these public. leave them private, and provide a constructor that takes byte[]. getSize() should simply give the length of the byte array. POJO doesnt mean it can't have _any_ logic!
The constructor should require a type and a byte[], and be immutable after that